### ISSUE TITLE
Installation instruction

### ISSUE THREAD

**Issue by Loran0** [GrandaddyShmax/audiocraft_plus#21]
Hello there

Can you please provide additional instructions on how can i install this awesome webUI if i've already got installed the latest version on audiocraft with all dependencies etc.
For example, i've got D:\audiocraft folder. Can i install PLUS in the same one? Or better to do D:\PLUS ? And the next steps?

Thank you for your help.

**Comment by GrandaddyShmax** [comment]
Sadly i myself cannot guide you how to install it on your pc as I myself do not have it installed since my GPU isn't powerful enough.
however i did find a video with explanation on how to do it, might help you out:
https://www.youtube.com/watch?v=WjGk4bcbUOI

I might be wrong but I think you will need to create a separate folder for audiocraft plus as it does changes quite few stuff, again cannot say for sure as i have not tested it myself.
Do let me know if it worked out for you ☺

**Comment by Loran0** [comment]
I already did install audiocraft according to your awesome instruction - https://github.com/FurkanGozukara/Stable-Diffusion/blob/main/Tutorials/AI-Music-Generation-Audiocraft-Tutorial.md

But now i'm trying to figure out how to install plus webUI. Reading your installation instr:
pip install -U audiocraft  # <-already done. It means open CMD (yep, using Win11) and did this:

`git clone https://github.com/facebookresearch/audiocraft`
`cd audiocraft`

Now i'm in d:\audiocraft

`pip install -e .`
`pip3 install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118`

Thats it, Now i'm running .\demos\musicgen_app.py and it works great with native UI.

Wanna figure out - do i even need to clone your repo? Like

`cmd -> git clone https://github.com/GrandaddyShmax/audiocraft_plus`

and the same steps like `cd audiocraft_plus` , pip install.... and so on. Do i need it? Or maybe i just need to put some files into my original audiocraft folder and just start .py to use PLUS WebUI?

my english sucks but i'm trying as i can :D

**Comment by GrandaddyShmax** [comment]
I think instead of cloning the original repo of audiocraft, just clone mine which is audiocraft plus. So you only need to clone a repo once
After that follow the exact steps and I believe it should work for you

**Comment by KannManMachen** [comment]
Hi there,
I'm just dealing with Python, libraries, etc. for the first time ... total beginner :D

A basic question: I had already seen that use is made of CUDA cores - can I get your WEB UI to run on MacOS at all or does my journey end here :)?

Thanks in advance

**Comment by Loran0** [comment]
i clone it...
after that i did everything according to this instr - https://github.com/FurkanGozukara/Stable-Diffusion/blob/main/Tutorials/AI-Music-Generation-Audiocraft-Tutorial.md

```
cd audiocraft_plus
python -m venv venv
cd venv
cd scripts
activate
...
```

So the last 2 string:
```
call .\venv\Scripts\activate.bat
python app.py --inbrowser
```

Only after all those steps PLUS WebUI works well.
I was just trying to make it easy for dummies like me... 

Btw if someone wanna make a shortcut  on your desktop, -make an empty file and put in it:

```
call "d:\\GitHub\\audiocraft_plus\\venv\\Scripts\\activate.bat"
python "d:\\GitHub\\audiocraft_plus\\demos\\musicgen_app.py"
```
d:\\GitHub\\audiocraft_plus   ---> put here your path to plus version.

for original audiocraft:

```
call "d:\\GitHub\\audiocraft\\venv\\Scripts\\activate.bat"
python "d:\\GitHub\\audiocraft\demos\\musicgen_app.py"
```

Then save it on your desktop or anywhere you want as .bat or .cmd

**Comment by GrandaddyShmax** [comment]
> Hi there, I'm just dealing with Python, libraries, etc. for the first time ... total beginner :D
> 
> A basic question: I had already seen that use is made of CUDA cores - can I get your WEB UI to run on MacOS at all or does my journey end here :)?
> 
> Thanks in advance

there was already a question related to Mac OS, check out this issue: https://github.com/GrandaddyShmax/audiocraft_plus/issues/15
in short, i made an additional branch for Mac OS called `mac-os-fix`, check it out and let me know if it works for you

**Comment by GrandaddyShmax** [comment]
> i clone it... after that i did everything according to this instr - https://github.com/FurkanGozukara/Stable-Diffusion/blob/main/Tutorials/AI-Music-Generation-Audiocraft-Tutorial.md
> 
> ```
> cd audiocraft_plus
> python -m venv venv
> cd venv
> cd scripts
> activate
> ...
> ```
> 
> So the last 2 string:
> 
> ```
> call .\venv\Scripts\activate.bat
> python app.py --inbrowser
> ```
> 
> Only after all those steps PLUS WebUI works well. I was just trying to make it easy for dummies like me...
> 
> Btw if someone wanna make a shortcut on your desktop, -make an empty file and put in it:
> 
> ```
> call "d:\\GitHub\\audiocraft_plus\\venv\\Scripts\\activate.bat"
> python "d:\\GitHub\\audiocraft_plus\\app.py"
> ```
> 
> d:\GitHub\audiocraft_plus ---> put here your path to plus version.
> 
> for original audiocraft:
> 
> ```
> call "d:\\GitHub\\audiocraft\\venv\\Scripts\\activate.bat"
> python "d:\\GitHub\\audiocraft\demos\\musicgen_app.py"
> ```
> 
> Then save it on your desktop or anywhere you want as .bat or .cmd

so in the end it works for you?

**Comment by Loran0** [comment]
> so in the end it works for you?

Yes. https://github.com/FurkanGozukara/Stable-Diffusion/blob/main/Tutorials/AI-Music-Generation-Audiocraft-Tutorial.md - This instruction i actually works on Windows also for plus version.
It's kinda weird coz a few days ago i use it only with

`python app.py`

Now i need to create an environment, activate and launch it. Dunno what changed in my system ... or in audiocraft repo :/

**Comment by KannManMachen** [comment]
> > Hi there, I'm just dealing with Python, libraries, etc. for the first time ... total beginner :D
> > A basic question: I had already seen that use is made of CUDA cores - can I get your WEB UI to run on MacOS at all or does my journey end here :)?
> > Thanks in advance
> 
> there was already a question related to Mac OS, check out this issue: #15 in short, i made an additional branch for Mac OS called `mac-os-fix`, check it out and let me know if it works for you

@GrandaddyShmax

Now it works like a charm <3 – thanks for the fast support!
